src/app/(sidebar)/endpoints/[[...pages]]/page.tsx

```diff
@@ -426,22 +539,33 @@ export default function Endpoints() {
   };
 
   const renderPostPayload = () => {
+    let renderedProps = getPostPayload();
+
     if (pageData?.requestMethod === "POST") {
+      if (pathname === Routes.ENDPOINTS_TRANSACTIONS_POST) {
+        renderedProps = { tx: params.tx ?? "" };
+      }
+
+      if (isRpcEndpoint) {
+        renderedProps = getRpcPostPayloadProps(pathname);
+      }
+    }
+
+    if (renderedProps) {
       return (
         <div className="Endpoints__txTextarea">
           <Textarea
             id="tx"
             fieldSize="md"
             label="Payload"
-            value={JSON.stringify({ tx: params.tx ?? "" }, null, 2)}
+            value={JSON.stringify(renderedProps, null, 2)}
```

Comment:
Just in case no condition is met, and we still end up with `undefined` here.

```suggestion
            value={renderedProps ? JSON.stringify(renderedProps, null, 2) : ""}
```